server/src/modules/queues/video-processor.js

```diff
@@ -102,4 +106,23 @@ const processMp4ToHls = async (filePath, outputFolder, jobData) => {
     return;
 };
 
-module.exports = { processRawFileToMp4, processMp4ToHls, generateThumbnail };
+const getVideoDuration = (filePath) => {
+
+    // if any error occour return 0.0 second as default video duration
+    // else return video duration
+    return new Promise((resolve,reject) => {
```

Comment:
I see you are using ffprobe, which is  a nice approach. You can also extract the `Resolution` information along with the `duration` so that we can save this as well. 

Resolve these two values like `{duration, resolution}`

I believe we may need to update our schema as well. 
Nice work

Reply:
vai, 

> I see you are using ffprobe, which is a nice approach. You can also extract the `Resolution` information along with the `duration` so that we can save this as well.
> 
> Resolve these two values like `{duration, resolution}`
> 
> I believe we may need to update our schema as well. Nice work

vai, along with `resolution` there might be other video metadata which might needed in future. So instead of returning `duration` & `resolution`, what if we return  `metadata` object and get any value we need  -

```
   const videoMetadata = await getVideoMetadata(`./${req.file.path}`)
   const videoDuration = parseInt(metadata.format.duration)
```

Reply:
I checked other metadata. I think we don't need those. 
And explicit direction is better than blackbox.

But you can give your counter argument as well. 👍

Reply:
> I checked other metadata. I think we don't need those. And explicit direction is better than blackbox.
> 
> But you can give your counter argument as well. +1

No vaia, currently i do not have any counter argument.

About `resolution`, what  would be the better way to store `resolution` ? Should we store as a string -
```
resolution = '{height}*{width}'
```
 or as an object -

```
resolution = {
    height : {height},
    width : {width}
}
```
I think we should store as an object. it'll make searching easier if we want to search for a specific height or width.

Reply:
Use the object format please.